lib/execution_engine2/utils/clients.py

```diff
@@ -87,33 +89,106 @@ class ClientSet:
     These are not user-specific and can be reused throughout the application.
     """
 
-    def __init__(self, cfg: Dict[str, str], cfg_path: str, debug: bool = False):
+    def __init__(
+        self,
+        auth: KBaseAuth,
+        auth_admin: AdminAuthUtil,
+        condor: Condor,
+        catalog_utils: CatalogUtils,
+        kafka_client: KafkaClient,
+        mongo_util: MongoUtil,
+        slack_client: SlackClient,
+    ):
         """
-        Initialize the client set from a configuration dictionary.
-
-        cfg - the configuration dictionary
-        cfg_path - the path to the configuration file
-        debug - set clients that support it to debug mode
-
-        Expected keys in config:
-        auth-url - the root URL of the kbase auth service
-        catalog-url - the URL of the catalog service
-        catalog-token - a token to use with the catalog service. Ideally a service token
-        kafka-host - the host string for a Kafka service
-        slack-token - a token for contacting Slack
+        Initialize the client set from the individual clients.
         """
-        # TODO seems like it'd make sense to init Condor from a config dict like everything else
-        self.condor = Condor(cfg_path)
-        self.catalog_utils = CatalogUtils(cfg["catalog-url"], cfg["catalog-token"])
-        auth_url = cfg["auth-url"]
-        self.auth = KBaseAuth(auth_url=auth_url + "/api/legacy/KBase/Sessions/Login")
-        # TODO using hardcoded roles for now to avoid possible bugs with mismatched cfg roles
-        #       these should probably be configurable
-        self.auth_admin = AdminAuthUtil(auth_url, [ADMIN_READ_ROLE, ADMIN_WRITE_ROLE])
-
-        # KafkaClient has a nice error message when the arg is None
-        self.kafka_client = KafkaClient(cfg.get("kafka-host"))
-        # SlackClient handles None arguments
-        self.slack_client = SlackClient(
-            cfg.get("slack-token"), debug=debug, endpoint=cfg.get("ee2-url")
-        )
+
+        # TODO check no clients are None. Make a general method somewhere
```

Comment:
There are a number of TODOs in the codebase I've left in because a) this PR is already 400+ lines, and b) they're not that important. I'll get to them in future PRs when I revisit these files.

Reply:
c) because they're way out of scope for this PR.